src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -760,16 +786,16 @@ SubCmdProgram::execute(const SubCmdOptions& _options) const
       throw xrt_core::error("Usage: xbmgmt program --device='0000:00:00.0' --base --image='/path/to/flash_image' OR shell_name");
 
     // We support up to 2 flash images 
-    if (image.size() > 2)
+    if (image.size() == 1)
```

Comment:
`If then else` ladders can be difficult to read and support especially with integer values.

In this case it would be better to have the following:
```
 if (image.size() == 1)
     auto_flash(workingDevice, image.front());

if (image.size() == 2) {
      std::cout << "CAUTION! Force flashing the platform on the device without any checks." <<
                   "Please make sure that the correct information is passed in." << std::endl;
      auto image_paths = find_flash_image_paths(image);
      update_shell(workingDevice->get_device_id(), flashType, image_paths.front(), (image_paths.size() == 2 ? image_paths[1]: ""));
    }

if (image.size() > 2)
    throw xrt_core::error("Please specify either 1 or 2 flash images");
```